src/authtoken/test.py

```diff
@@ -0,0 +1,226 @@
+        stamp = timezone.now() + datetime.timedelta(minutes=10)
+        token = jwt.encode(
+            {'exp': stamp, 'account_id': None},
+            'secret',
+            algorithm=self.enc
+        )
+        auth = 'CardToken {0}'.format(token.decode())
+        response = self.csrf_client.get(
+            self.path,
+            {'example': 'example'},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_auth_with_bad_header(self):
+        account = AccountFactory()
+        stamp = timezone.now() + datetime.timedelta(minutes=10)
+        token = jwt.encode(
+            {'exp': stamp, 'account_id': str(account.pk)},
+            'secret',
+            algorithm=self.enc
+        )
+        auth = 'ThisIsIncorrect {0}'.format(token.decode())
+        response = self.csrf_client.get(
+            self.path,
+            {'exp': stamp, 'account_id': str(account.pk)},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+        # Returns 200_OK as it is not allowed to even try to authenticate
+        # And there is no other authentication that is tested
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        auth = 'CardToken'
+        response = self.csrf_client.get(
+            self.path,
+            {'exp': stamp, 'account_id': str(account.pk)},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        auth = 'CardToken {0} {1}'.format(token.decode()[:10], token.decode())
+        response = self.csrf_client.get(
+            self.path,
+            {'exp': stamp, 'account_id': str(account.pk)},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+@override_settings(ROOT_URLCONF='authtoken.test', SECRET_WEBTOKEN='secret')
+class CreateFooCardTokenTests(TestCase):
+    enc = 'HS256'
+    path = '/auth-token/'
+
+    def setUp(self):
+        self.csrf_client = APIClient(enforce_csrf_checks=True)
+
+    def test_create_new_account_token(self):
+        card = CardFactory()
+        api_token = Token.objects.create(key='abc123')
+        auth = 'Token {0}'.format(api_token)
+        response = self.csrf_client.post(
+            self.path,
+            {'number': card.number},
+            format='json',
+            HTTP_AUTHORIZATION=auth
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn('card_token', response.data.keys())
+
+        token = response.data.get('card_token')
+        self.assertIsNotNone(token)
+        self.assertTrue(len(token) > 0)
+
+        try:
+            decoded = jwt.decode(token, 'secret', algorithm=self.enc)
```

Comment:
Should this be done manually like this? Shouldn't be enough to test if you can authenticate with a token that you got from the /auth-token/ endpoint?